pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -93,25 +93,38 @@ func NewEndpointController(ctx context.Context, podInformer coreinformers.PodInf
 	}
 
 	serviceInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
-		AddFunc: e.onServiceUpdate,
+		AddFunc: func(obj interface{}) {
+			e.onServiceUpdate(ctx, obj)
```

Comment:
Is this correct? Is the context of `NewEndpointController` correct to use for handlers that will be invoked asynchronously later?

(I don't know the answer to this.)

Reply:
I updated it to pass around logger. It's already being done in some parts of code like [here](https://github.com/kubernetes/kubernetes/blob/d39b162163006799cc8e0198fa83d0b6d36a41b9/pkg/controller/endpointslice/endpointslice_controller.go#L147C6-L147C27).